### PR TITLE
[Localization] Aries

### PR DESCRIPTION
**Pull request by JavierMatosD**
This PR localizes messages in the following files:
* export.cpp
* export.ifw.cpp
* export.prefab.cpp

### CODE REVIEW

File: include/vcpkg/base/messages.h
Comment (Thomas1664) [inline include/vcpkg/base/messages.h:1747]:
```suggestion
    DECLARE_MESSAGE(WritingManifest, (), "", " [DEBUG] Writing manifest to");
```

File: src/vcpkg/export.cpp
Comment (Thomas1664) [inline src/vcpkg/export.cpp:485]:
```suggestion
        msg::println(msg::format(msgUsingExportedLibs).append_indent().append_raw(cmake_variable.s).append_raw("\n"));
```

`printnl` already adds a `\n`

File: src/vcpkg/export.prefab.cpp
Comment (Thomas1664) [inline src/vcpkg/export.prefab.cpp:618]: You are removing one `\n` here.

File: src/vcpkg/export.cpp
Comment (Thomas1664) [inline src/vcpkg/export.cpp:109]:
```suggestion
                    msg::println(msg::format(msgExportingPackages).append_raw("\n" + as_string));
```

`println` already adds `\n`. Same below.

File: src/vcpkg/export.cpp
Comment (Thomas1664) [inline src/vcpkg/export.cpp:568]: You're removing one `\n` here.

File: src/vcpkg/export.prefab.cpp
Comment (Thomas1664) [inline src/vcpkg/export.prefab.cpp:216]: You're removing one `\n` here.

File: src/vcpkg/export.prefab.cpp
Comment (Thomas1664) [inline src/vcpkg/export.prefab.cpp:493]:
```suggestion
                                 .append_raw('\n')
```

Quick perf tip: Saves call to `strlen`. Same elsewhere.

File: src/vcpkg/export.prefab.cpp
Comment (Thomas1664) [inline src/vcpkg/export.prefab.cpp:496]:
```suggestion
```

Already added by `printnl`.

File: include/vcpkg/base/messages.h
Comment (JavierMatosD) [reply include/vcpkg/base/messages.h:927]: Good catch :)

**Comment by JavierMatosD** [comment]
Thanks for the review @Thomas1664 :)

File: src/vcpkg/export.prefab.cpp
Comment (JavierMatosD) [reply src/vcpkg/export.prefab.cpp:216]: This was intentional

File: src/vcpkg/export.cpp
Comment (JavierMatosD) [reply src/vcpkg/export.cpp:568]: This was intentional

File: src/vcpkg/export.prefab.cpp
Comment (JavierMatosD) [reply src/vcpkg/export.prefab.cpp:618]: This was intentional

File: src/vcpkg/export.prefab.cpp
Comment (BillyONeal) [inline src/vcpkg/export.prefab.cpp:32]: Should this call `exit_filesystem_call_error`? I'm not sure the "to find modules" part is user meaningful.

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:986]: Ditto --prefab-debug

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:989]: Ditto "print before we do the thing" and no need to say we did the thing.

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1069]:
```suggestion
    DECLARE_MESSAGE(MutuallyRequiredOption,
                    (msg::value, msg::option),
                    "{value} is a second {option} switch",
                    "--{value} must be used with --{option}.");
```

which leaves room for this one either here or in a separate review:

```
    DECLARE_MESSAGE(MutuallyExclusiveOption,
                    (msg::value, msg::option),
                    "{value} is a second {option} switch",
                    "--{value} can not be used with --{option}.");
```

Should we include the -- in the `{value}`? That way we can say exactly what the user typed if we are accepting some other form, although that may be at most a theoretical concern.

As an aside, should we always say "switch" even though internally we distinguish between switch/option/multi-option?

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1094]: Ditto --debug-prefab

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1113]: I think this one needs to be combined with `PrebuiltPackages` to be meaningfully localized.

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1298]:
```suggestion
    DECLARE_MESSAGE(MissingAndroidEnv, (), "", "ANDROID_NDK_HOME environment variable missing");
```

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1310]: I think this error is unactionable but it's what the old code did so no change required.

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1357]: I think this should reuse `CreatedNuGetPackage`. Or do the "creating NuGet package {path}" transform I mentioned before and fix that place too.

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1377]: I think we should have a generic "your vcpkg "installed" tree is corrupted" message and this should use that.

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1388]: I think we should standardize in "creating" for this.

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1405]: Ditto --prefab-debug

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1446]:
```suggestion
    DECLARE_MESSAGE(ProvideExportType,
                    (),
                    "",
                    "At least one of the following options are required: --raw --nuget --ifw --zip --7zip --chocolatey --prefab.");
```

File: src/vcpkg/export.ifw.cpp
Comment (BillyONeal) [inline src/vcpkg/export.ifw.cpp:426]: I think this should do:

```
fs.remove_all(ifw_packages_dir_path, VCPKG_LINE_INFO);
```

and delete `RemovingPackageDirFailed`.

File: src/vcpkg/export.ifw.cpp
Comment (BillyONeal) [inline src/vcpkg/export.ifw.cpp:429]: Ditto I think this should call `fs.create_directory(ifw_packages_dir_path, VCPKG_LINE_INFO)`.

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1516]: @ras0219-msft Is this saying "checkout" as in "git checkout", or is it saying "check out" as in "look at"? If the latter I think it should say:

```
Exported {package_name} to {path}
```

or similar.

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1548]: I think this one should get expected and actual instead of hard coding 15.

File: src/vcpkg/export.cpp
Comment (BillyONeal) [inline src/vcpkg/export.cpp:485]:
```
    DECLARE_MESSAGE(CMakeUsingExportedLibs, (), "{value} is a CMake command line switch of the form -DFOO=BAR", "To use exported libraries in CMake projects, add {value} to your CMake command line.");
```

```suggestion
        msg::println(msg::format(msgCMakeUsingExportedLibs, msg::value = cmake_variable.s));
```

I think including the bit in the message makes it more meaningful to localizers.

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1749]: Ditto --prefab-debug

File: include/vcpkg/base/messages.h
Comment (JavierMatosD) [reply include/vcpkg/base/messages.h:752]: Ah, nice catch! I'll split the message :)

File: include/vcpkg/base/messages.h
Comment (JavierMatosD) [reply include/vcpkg/base/messages.h:759]: I am not sure what you mean. Can you elaborate?

File: src/vcpkg/export.ifw.cpp
Comment (JavierMatosD) [reply src/vcpkg/export.ifw.cpp:426]: So rather than using the check exit, 
```
if (ec) 
{
    fs.remove_all(ifw_packages_dir_path, VCPKG_LINE_INFO);
}
```
?

Comment (BillyONeal) [reply src/vcpkg/export.ifw.cpp:424]: No, just:

```
fs.remove_all(ifw_packages_dir_path, VCPKG_LINE_INFO);
```

There's no need for anything related to EC when using the LineInfo overloads of the filesystem things; they already terminate with a message on failure.

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [reply include/vcpkg/base/messages.h]: We have a bunch of different patterns here, with different error message strategies, etc:

```
* Creating <type>...
* created <path>.
and
* Creating <type> <path>...
* done.
and
* (nothing)
* Created <type> <path>.
and
* Creating <type>
* Created <type> <path>.
```

I'm saying I think we should change all such patterns to be the same. My suggestion is:

```
* Creating <type> <path>...
```

with no "done" message. (The "done" is implied by the next "Creating" message or by the program terminating)

This reduces the number of localized messages we need per <type> to 1. So you'd have:

```
DECLARE_MESSAGE(CreatingNuGetPackage, (msg::path), "", "Creating nupkg \"{path}\"...");
```

Comment (JavierMatosD) [reply include/vcpkg/base/messages.h]: I see; in that case, is it OK to hold off on this kind of refactoring until all the PRs are merged? That way, I can avoid merge conflicts as much as possible and can look at all the messages together :)

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [reply include/vcpkg/base/messages.h:816]: Up that to 100%. All these "foo plan cannot be empty" are detecting bugs in the dependency resolver.

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [reply include/vcpkg/base/messages.h]: Yes, that's what I meant by:

>I would really like to see the consistent "creating {path}" treatment done here as even this one command isn't consistent about where it puts and does not put paths, but I would understand if you were like "Bill, this is just trying to localize what we have, not trying to fix the world"

The argument to do it now is that it avoids wasted effort on the part of our localizers because half the reason to choose this form is to reduce the total number of localized messages.

File: include/vcpkg/base/messages.h
Comment (vicroms) [reply include/vcpkg/base/messages.h:1642]: Agree with Billy here, we should simplify the message.

File: include/vcpkg/base/messages.h
Comment (vicroms) [inline include/vcpkg/base/messages.h:566]: I feel like this message could be simplified.

```suggestion
                    "export prefab requires a triplet that sets CMAKE_SYSTEM_NAME to \"android\"");
```

Or maybe just: `export prefab requires an Android triplet.`.

**Review by vicroms — APPROVED** [review]

File: src/vcpkg/remove.cpp
Comment (BillyONeal) [inline src/vcpkg/remove.cpp:262]: That seems wrong; I think you meant to add the additional packages to export message rather than replace the packages to remove one.

File: src/vcpkg/export.prefab.cpp
Comment (BillyONeal) [inline src/vcpkg/export.prefab.cpp:651]:
```suggestion
                Debug::println(
```

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:562]:
```suggestion
    DECLARE_MESSAGE(ExportPrefabRequiresAndroidTriplet, (), "", "export prefab requires an Android triplet.");
``` ?

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:765]:
```suggestion
    DECLARE_MESSAGE(PackingNugetPackage, (), "", "Packing NuGet package...");
```

I think `nuget pack` is the command and I think we should be specific about it? Either that or the message should be changed to "creating" to match.

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1174]:
```suggestion
    DECLARE_MESSAGE(InstallingMavenFile, (msg::path), "Printed after a filesystem operation error", "{path} installing maven file");
```

File: include/vcpkg/base/messages.h
Comment (BillyONeal) [inline include/vcpkg/base/messages.h:1880]: This one is not sorted

**Review by BillyONeal — APPROVED** [review]
This approval is contingent on the damage to `remove` being reverted.

File: include/vcpkg/base/messages.h
Comment (JavierMatosD) [reply include/vcpkg/base/messages.h:1880]: I know. I will sort it at the end to minimize conflicts :)

**Review by BillyONeal — APPROVED** [review]

**Pull request merged**
